src/sync/sync_session.hpp

```diff
@@ -49,6 +49,7 @@ namespace sync {
 class Session;
 }
 
+using SessionWaiterPointer = void(sync::Session::*)(std::function<void(std::error_code)>);
```

Comment:
Since this is only used internally it'd be better to put this inside the class, alongside the `private` members.